### PR TITLE
Week 3 conrad

### PR DESCRIPTION
**Pull request by charper95**

### CODE REVIEW

File: react-debugging.html [hunk 1]
```diff
@@ -55,6 +55,7 @@ <h1>Draw a Card</h1>
               <Card
                 drawCard={this.drawCard}
                 shuffle={this.shuffle}
+                hand={this.state.hand} // In order for the Card component to call on hand key in state it needs to be assigned to a variable and passed into the Card component call so the Card component can use it as props.
```
Comment (WalkerTexas-coder) [inline react-debugging.html:58]: Good catch!

File: react-debugging.html [hunk 2]
```diff
@@ -77,12 +78,12 @@ <h1>Draw a Card</h1>
             margin: "10px",
             padding: "10px"
           }
-          return(
+          return( // needed to call on the drawCard & shuffle methods by using this.props
             <>
-              <button onClick={}>
+              <button onClick={this.props.drawCard}>
```
Comment (WalkerTexas-coder) [inline react-debugging.html:83]: Yes!

File: react-debugging.html [hunk 3]
```diff
@@ -77,12 +78,12 @@ <h1>Draw a Card</h1>
...
                 Click to Draw a Card
               </button>
-              <button onClick={shuffle}>
+              <button onClick={this.props.shuffle}>
```
Comment (WalkerTexas-coder) [inline react-debugging.html:86]: Good!

File: interview-questions.md [hunk 4]
```diff
@@ -9,59 +9,59 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
-  Researched answer:
+  Researched answer: A DOM event is any input by a user on the website. When the user changes anything, such as clicking or hovering, that is a DOM event. The DOM reacts to changes and responds by changing what the user sees depending on the type of event. In JSX there are event listeners that listen for specified interactions on a node. An example of an event listener is onClick that listens for a click event. DOM events are shortened to "e".
 
 
 
 6. STRETCH: What is an anonymous function in JavaScript?
 
-  Your answer:
+  Your answer: An anonmyous function is a function that does not have a name.
```
Comment (WalkerTexas-coder) [inline interview-questions.md:52]: Well done!

File: interview-questions.md [hunk 5]
```diff
@@ -9,59 +9,59 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
 
 
 5. What is a DOM event?
 
-  Your answer:
+  Your answer: A DOM event is any input by a user on the website. When the user changes anything, such as clicking or hovering, that is a DOM event.
 
-  Researched answer:
+  Researched answer: A DOM event is any input by a user on the website. When the user changes anything, such as clicking or hovering, that is a DOM event. The DOM reacts to changes and responds by changing what the user sees depending on the type of event. In JSX there are event listeners that listen for specified interactions on a node. An example of an event listener is onClick that listens for a click event. DOM events are shortened to "e".
```
Comment (WalkerTexas-coder) [inline interview-questions.md:46]: Great answer here. especially getting to the final point where you show you know what the implications are for this idea in your code.

File: interview-questions.md [hunk 6]
```diff
@@ -9,59 +9,59 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
 
 
 3. What are props in React?
 
-  Your answer:
+  Your answer: Props is a term in React that is short for properties. Props allow developers to pass data between components in React.
 
-  Researched answer:
+  Researched answer: Props is a term in React that is short for properties. Props allow developers to pass data down into nested components. Data is passed down in only one direction from a parent component to a child component. Props pass data from the state object of a component. Props can't be updated or change because they are read-only. Props are called using this.props.
```
Comment (WalkerTexas-coder) [inline interview-questions.md:30]: Nice!

File: code-challenges.test.js [hunk 7]
```diff
@@ -18,46 +22,138 @@
...
+    let newArray = array.concat()
+// Iterate through the given array starting at index 1 because index 0 should remain the same
+    for (let i = 1; i < array.length; i++) {
+// Assign the current index of the newArray (starting from index 1) to the sum of the previous index and the current index
+        newArray[i] = newArray[i-1] + array[i]
+    }
+// After iterating through and replacing each index, return the changed array
+    return newArray
+}
+
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js:151]: Dude! Excellent!

File: code-challenges.test.js [hunk 8]
```diff
@@ -18,46 +22,138 @@
...
+    let newArray = array.filter(value => {
+        if (value % 2 !== 0 && typeof value === "number") {
+            return value
+        }
+    })
+// Use the .sort() method to sort the array from least to greatest
+// The .sort() method must contain a compareFunction as a parameter, if it is not provided the elements of the array are converted into strings and sorted alphabetically
+// The compareFunction takes two parameters, the first element and the second element, in order to sort the numbers in ascending order you have to subtract the second element from the first element
+    return newArray.sort((a, b) => a - b)
+}
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js:107]: Cleeeeean!

File: code-challenges.test.js [hunk 9]
```diff
@@ -18,46 +22,138 @@
...
+// And you want to take index 0 of the array which is 2-2=0. The value of the 0th index being 1
+            let array2 = resultArray[i - 2]
+// And you want to take values of those indexes and add them together. 1+1=2
+// To finalize, push the result into the resultArray
+            resultArray.push(array1 + array2)
+        }
+// Return the result
+        return resultArray
+    }
+}
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js:64]: Readable, great pseudo code, thoughtfully applied edge case scenarios!

**Review by WalkerTexas-coder — APPROVED** [review]
Conrad great job! This is an excellent assessment. I really appreciate that you have your pseudo code associated with the code it represents on the line below it! Your interview questions are really good. Well done elaborating on the research answers. Your PR is approved!
